bundles/org.openhab.core.voice/src/main/java/org/openhab/core/voice/internal/VoiceManagerImpl.java

```diff
@@ -598,13 +603,24 @@ public void listenAndAnswer(@Nullable STTService stt, @Nullable TTSService tts,
             throw new IllegalStateException(
                     "Cannot execute a simple dialog as provided locale is not supported by all services.");
         } else {
-            DialogProcessor processor = dialogProcessors.get(audioSource.getId());
-            if (processor == null) {
+            var isSingleDialog = false;
+            DialogProcessor activeProcessor = dialogProcessors.get(audioSource.getId());
+            cleanSingleDialogProcessors();
+            if (activeProcessor == null) {
+                isSingleDialog = true;
+                activeProcessor = singleDialogProcessors.get(audioSource.getId());
+            }
+            var processor = new DialogProcessor(sttService, ttsService, prefVoice, interpreters, audioSource, audioSink,
+                    loc, item, this.eventPublisher, this.i18nProvider, b);
+            if (activeProcessor == null) {
                 logger.debug("Executing a simple dialog for source {} ({})", audioSource.getLabel(null),
                         audioSource.getId());
-                processor = new DialogProcessor(sttService, ttsService, prefVoice, interpreters, audioSource, audioSink,
-                        loc, item, this.eventPublisher, this.i18nProvider, b);
                 processor.start();
+                singleDialogProcessors.put(audioSource.getId(), processor);
+            } else if (!isSingleDialog && activeProcessor.isCompatible(processor)) {
+                logger.debug("Executing a simple dialog for active source {} ({})", audioSource.getLabel(null),
+                        audioSource.getId());
+                activeProcessor.startSingleDialog();
```

Comment:
This will interrupt a potential running dialogue.
And I am not sure the permanent dialogue will be restored after ?

Reply:
Yes, it's safe, the SpotEvent of the ks also calls this method under the hood, so it works well, it ignore other SpotEvent in the mean time but it continue with the spotting after it's finish.